src/tiled/objecttemplatemodel.cpp

```diff
@@ -153,6 +153,24 @@ bool ObjectTemplateModel::addNewDocument(TemplateGroupDocument *document)
     return true;
 }
 
+bool ObjectTemplateModel::addDocument(TemplateGroupDocument *document)
+{
+    if (!document)
```

Comment:
Yes, because it is simply invalid to try adding null document, so it's never expected to happen. The assert would document that and in addition checks for it in debug mode.